language/move-prover/spec-lang/src/translate.rs

```diff
@@ -2686,6 +2678,24 @@ impl<'env, 'translator> ModuleTranslator<'env, 'translator> {
             std::mem::take(&mut self.instantiation_map),
             std::mem::take(&mut self.spec_block_infos),
         );
+        // Warn about unused schemas.
+        for name in &self.parent.unused_schema_set {
+            let entry = self
+            .parent
+            .spec_schema_table
+            .get(&name)
+            .expect("schema defined");
+            let schema_name = name.display_simple(self.symbol_pool()).to_string();
+            let module_env = self.parent.env.get_module(entry.module_id);
```

Comment:
This unfortunately doesn't work, because the translator builds the module_env's incrementally up in the GlobalEnv. That is if you have modules 1..N and you are translating module N, then the environment only contains modules 1..(N-1). The result of translation is to add module N to the environment. (For background, 1..N are organized in topological use order.)

Reply:
But this is at the end of module translation inside function `populate_env_from_result` and after the module has been added to the global env(on line 2667). So module N should be available in the environment at this point?